Description: Fix build failure with the Debian wheezy kernel which backports the drm subsystem from Linux 3.4.
Bug-Debian: http://bugs.debian.org/703358
Author: Felix Geyer <fgeyer@debian.org>

Index: virtualbox/src/VBox/Additions/linux/drm/vboxvideo_drm.c
===================================================================
--- virtualbox.orig/src/VBox/Additions/linux/drm/vboxvideo_drm.c
+++ virtualbox/src/VBox/Additions/linux/drm/vboxvideo_drm.c
@@ -97,6 +97,12 @@
 # endif
 #endif
 
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 2, 0)
+#  ifdef DRM_IOCTL_MODE_GETPLANE
+#   define DRM_DEBIAN70
+#  endif
+# endif
+
 static struct pci_device_id pciidlist[] = {
         vboxvideo_PCI_IDS
 };
@@ -109,7 +115,7 @@
     return 0;
 #endif
 }
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0) || defined(DRM_FOPS_AS_POINTER)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0) || defined(DRM_FOPS_AS_POINTER) || defined(DRM_DEBIAN70)
 /* since linux-3.3.0-rc1 drm_driver::fops is pointer */
 static struct file_operations driver_fops =
 {
@@ -134,7 +140,7 @@
     .get_map_ofs = drm_core_get_map_ofs,
     .get_reg_ofs = drm_core_get_reg_ofs,
 #endif
-# if LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0) && !defined(DRM_FOPS_AS_POINTER)
+# if LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0) && !defined(DRM_FOPS_AS_POINTER) && !defined(DRM_DEBIAN70)
     .fops =
     {
         .owner = THIS_MODULE,
@@ -150,7 +156,7 @@
         .mmap = drm_mmap,
         .poll = drm_poll,
     },
-#else /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0) || defined(DRM_FOPS_AS_POINTER) */
+#else /* LINUX_VERSION_CODE >= KERNEL_VERSION(3, 3, 0) || defined(DRM_FOPS_AS_POINTER) || defined(DRM_DEBIAN70) */
     .fops = &driver_fops,
 #endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 39) && !defined(DRM_RHEL61)
